### PR TITLE
Refactor out SquareClient to services module

### PR DESCRIPTION
**Pull request by Yong-L**
Per rails best practices, we should be using `services` convention

https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial

### CODE REVIEW

File: app/controllers/charges_controller.rb [hunk 1]
```diff
@@ -107,34 +107,27 @@ def validate(seller_id:, line_items:)
   def generate_description(seller_name:, item_types:)
     description = 'Thank you for supporting ' + seller_name + '.'
     if item_types.include? 'gift_card'
-      description +=
-        ' Your gift card(s) will be emailed to you when the seller opens back up.'
+      description += ' Your gift card(s) will be emailed to you when the seller opens back up.'
```
Comment (jtmckibb) [inline app/controllers/charges_controller.rb:110]: Can you run `rubocop -a` to make sure this line isn't too long?

File: app/controllers/charges_controller.rb [hunk 2]
```diff
@@ -107,34 +107,27 @@ def validate(seller_id:, line_items:)
...
-      source_id: source_id,
-      idempotency_key: SecureRandom.uuid,
-      amount_money: { amount: amount, currency: 'USD' },
-      buyer_email_address: email,
-      note: note,
-      location_id: square_location_id
-    }
-
-    api_response = api_client.payments.create_payment(body: request_body)
+    api_response = SquareManager::PaymentCreator.call(
```
Comment (jtmckibb) [inline app/controllers/charges_controller.rb:122]: Wow this is so much cleaner

File: app/controllers/charges_controller.rb (hunk 1 above)
Comment (Yong-L) [reply app/controllers/charges_controller.rb]: Just broke it up. I'm not too sure what the best practice is for ruby for breaking up lines haha

Comment (jtmckibb) [reply app/controllers/charges_controller.rb]: My own preference is to try and keep the string together so that it's easier to grep — but idc really lol

Comment (jtmckibb) [reply app/controllers/charges_controller.rb]: usually I just ignore it if it's just a long string

Comment (Yong-L) [reply app/controllers/charges_controller.rb]: Only problem was that even with the previous change the line was at 82 which exceeded the default 80. Maybe we should change the default limit to 100 or so

**Pull request merged**